### ISSUE TITLE
A false negative about the rule SimplifiableTestAssertion

### ISSUE THREAD

**Issue by Jane-PL-256** [pmd/pmd#4062]
**Affects PMD Version:6.47.0** 

**Rule:SimplifiableTestAssertion**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.47.0/pmd_rules_java_bestpractices.html#simplifiabletestassertion>

**Description:**
Hi, I found a false negative about the rule SimplifiableTestAssertion, please read the following code example. In line 6, PMD should have reported a warning because this line can be simplified into `assertEquals(a, b);`, but it doesn't. Hence, I think this is a false negative. Thanks.

**Code Sample demonstrating the issue:**
```java
public class A {
    class B {
        Object a, b;
        @Test
        public void test() {
            assertTrue(a.equals(b));  // should report a warning in this line
        }
    }
}
```

**Expected outcome:**

PMD should report a violation at line 6, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0.
